### PR TITLE
(feat): Display Project Settings Inbound Filters

### PR DESCRIPTION
**Pull request by LSSangha**
Users can view the changes made within the project setting's inbound filters. 
Screen Shot 2019-04-24 at 4 33 42 PM

### CODE REVIEW

File: src/sentry/api/endpoints/project_filter_details.py [hunk 1]
```diff
@@ -26,6 +27,36 @@ def put(self, request, project, filter_id):
         if not serializer.is_valid():
             return Response(serializer.errors, status=400)
 
-        filter.enable(serializer.object)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
```
Comment (markstory) [inline src/sentry/api/endpoints/project_filter_details.py]: Don't we also have a `not serializer.is_valid()` on line 27?

File: src/sentry/api/endpoints/project_filter_details.py [hunk 2]
```diff
@@ -26,6 +27,36 @@ def put(self, request, project, filter_id):
...
+            added = new_state - current_state
+            removed = current_state - new_state
+
+            if removed:
+                returned_state = removed
+                audit_log_state = AuditLogEntryEvent.PROJECT_DISABLE
+            elif added:
+                returned_state = added
+
+        if filter.id == 'browser-extensions' or filter.id == 'localhost' or filter.id == 'web-crawlers':
```
Comment (markstory) [inline src/sentry/api/endpoints/project_filter_details.py]:
```suggestion
        if filter.id in ('browser-extensions', 'localhost', 'web-crawlers'):
```

Using `in` will save you some words.

File: src/sentry/models/auditlogentry.py [hunk 3]
```diff
@@ -260,6 +264,16 @@ def get_note(self):
             return 'requested to transfer project %s' % (self.data['slug'], )
         elif self.event == AuditLogEntryEvent.PROJECT_ACCEPT_TRANSFER:
             return 'accepted transfer of project %s' % (self.data['slug'], )
+        elif self.event == AuditLogEntryEvent.PROJECT_ENABLE:
+            if isinstance(self.data['state'], set):
+                return 'enabled project filter %s' % (', '.join(self.data["state"]),)
+            else:
+                return 'enabled project filter %s' % (self.data['state'], )
+        elif self.event == AuditLogEntryEvent.PROJECT_DISABLE:
+            if isinstance(self.data['state'], set):
```
Comment (markstory) [inline src/sentry/models/auditlogentry.py]: Do you need to handle lists or tuples?

Comment (LSSangha) [reply src/sentry/models/auditlogentry.py]: Lists

Comment (markstory) [reply src/sentry/models/auditlogentry.py]: Should this join case also be run for lists then? If your check was

```python
if isinstance(self.data['state'], six.string_types):
   return 'disabled project filter %s' % (self.data['state'], )
return 'disabled project filter %s' % (', '.join(self.data["state"]),)
```
You could handle strings, lists and other enumerables with fewer typechecks.

**Review by markstory — APPROVED** [review]

**Pull request merged**